src/helpers.rs

```diff
@@ -428,11 +435,14 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 }
             })?
         } else {
-            // FIXME: we have to implement the Windows equivalent of this.
-            throw_unsup_format!(
-                "setting the last OS error from an io::Error is unsupported for {}.",
-                target.target_os
-            )
+            // FIXME: we have to finish implementing the Windows equivalent of this.
+            this.eval_windows(match e.kind() {
+                NotFound => "ERROR_FILE_NOT_FOUND",
+                _ => throw_unsup_format!(
+                    "setting the last OS error from an io::Error is yet unsupported for {}.",
+                    target.target_os
```

Comment:
This error makes no sense here: the target OS is fine (it's Windows), but the error is the issue. So the error should be like `io error {} cannot be transformed into a raw os error` above.